product_sequence/models/product_category.py

```diff
@@ -39,20 +41,65 @@ def _prepare_ir_sequence(self, prefix):
     @api.multi
     def write(self, vals):
         prefix = vals.get("code_prefix")
-        if prefix:
-            for rec in self:
-                if rec.sequence_id:
-                    rec.sudo().sequence_id.prefix = prefix
-                else:
+        for rec in self:
+            if vals.get("parent_id"):
+                categories = self.env["product.category"].search(
+                    [("id", "=", vals["parent_id"])], limit=1
+                )
+                for category in categories:
+                    if category.parent_id.sequence_id:
+                        sequence = category.parent_id.sequence_id
+                        vals["code_prefix"] = sequence.prefix
+                        vals["sequence_id"] = sequence.id
+                    else:
+                        vals["code_prefix"] = categories.sequence_id.prefix
+                        vals["sequence_id"] = categories.sequence_id.id
+            elif rec.parent_id:
+                vals["code_prefix"] = rec.parent_id.sequence_id.prefix
+                vals["sequence_id"] = rec.parent_id.sequence_id.id
+            else:
+                if prefix is False:
+                    vals["sequence_id"] = ""
+                elif prefix:
                     seq_vals = self._prepare_ir_sequence(prefix)
-                    rec.sequence_id = self.env["ir.sequence"].create(seq_vals)
+                    vals_sequence = self.env["ir.sequence"].search(
+                        [("code", "=", seq_vals["code"])]
+                    )
+                    if bool(vals_sequence) is False:
+                        rec.sequence_id = self.env["ir.sequence"].create(seq_vals)
+                    else:
+                        rec.sequence_id = vals_sequence
+                    childs = self.env["product.category"].search(
+                        [("id", "child_of", rec.id)]
+                    )
+                    for child in childs:
+                        child.sequence_id = rec.sequence_id
```

Comment:
Aqui en vez de un for deberías poder hacer un write del campo directamente.

```suggestion
                    childs.write({'sequence_id':  rec.sequence_id.id, 'code_prefix': prefix})
```